### PR TITLE
Port accessibility fix for the text view control in lightbulb preview

### PR DESCRIPTION
**Pull request by mavasani**
Addresses [AB#1832532](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1832532)

This PR ports the LSP editor [accessibility fix](https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/496210) to the Roslyn preview pane implementation. We now show an interactive preview, which has a left border showing `+` and `-` for the changes, and keyboard navigation and text selection are now enabled for the preview text view.

## Before (non-interactive)
NonInteractivePreview

## After (interactive)
InteractivePreview

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.NavigationalCommandTarget.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal sealed partial class DifferenceViewerPreview
+    {
+        private sealed class NavigationalCommandTarget(ITextView textView, IEditorOperations editorOperations) : IOleCommandTarget
```
Comment (mavasani) [inline src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.NavigationalCommandTarget.cs:15]: All the code in this file is a port from https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/496210

File: src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.cs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Differencing;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal sealed partial class DifferenceViewerPreview : IDifferenceViewerPreview<IWpfDifferenceViewer>
```
Comment (mavasani) [inline src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.cs:22]: Almost all the code in this file is a port from https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/496210, while also retaining the prior code in https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core/Preview/DifferenceViewerPreview.cs

File: src/VisualStudio/Core/Def/PreviewPane/PreviewPane.xaml [hunk 3]
```diff
@@ -140,7 +140,7 @@
             <Separator Name="HeaderSeparator" Margin="0" Visibility="Collapsed"/>
         </StackPanel>
         <DockPanel Name="PreviewDockPanel" DockPanel.Dock="Top" Visibility="Collapsed">
-            <ScrollViewer Name="PreviewScrollViewer" IsTabStop="True"
+            <ScrollViewer Name="PreviewScrollViewer" IsTabStop="False"
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/PreviewPane/PreviewPane.xaml:143]: This tab stop is now moved to the text view control host

**Review by olegtk — APPROVED** [review]

**Comment by mavasani** [comment]
@dotnet/roslyn-ide can I please get another review?

**Review by arkalyanms — APPROVED** [review]

**Comment by mavasani** [comment]
Thanks!

File: src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.NavigationalCommandTarget.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.NavigationalCommandTarget.cs:15]: i really hate this.  is there any plan to make a reusable component here?  this seems to be taking us backwards wrt to being able to use the editor cleanly.

**Comment by CyrusNajmabadi** [comment]
@olegtk is this just a point in time thing, until we have a reusable difference viewer with the behavior needed? or is the expectation that roslyn always needs to own and maintain all this code?

**Comment by olegtk** [comment]
@CyrusNajmabadi yes, this is a tactical fix to satisfy October 6 SLA for this accessibility bug. Yes, we will provide an interactive diff viewer control in 17.9 to replace this.

**Comment by CyrusNajmabadi** [comment]
ok.  then i'm ok with this :)  thanks  @olegtk .  @mavasani can you file issue for tracking this?

**Comment by mavasani** [comment]
> ok. then i'm ok with this :) thanks @olegtk . @mavasani can you file issue for tracking this?

#70165

**Comment by olegtk** [comment]
on our side it's tracked by https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1893411, so far scheduled for 17.9 Preview1.

**Pull request merged**